opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -391,6 +392,11 @@ class WellModelMatrixAdapter : public Dune::AssembledLinearOperator<M,X,Y>
             // Construct scalar product.
             auto sp = Dune::createScalarProduct<Vector,POrComm>(parallelInformation_arg, category);
 
+            // If overlapLayers_ > 1 we are using Restricted Additive Schwarz and therefore need
+            // the residual at overlap DoFs.
+            if (overlapLayers_ > 1)
+                parallelInformation_arg.copyOwnerToAll(istlb, istlb);
```

Comment:
Please explain why this is only needed for more than one overlap layers. Shouldn't the right hand side always be consistent (all procs have the same values?)

Reply:
The right hand side is normally unique, which is correct with only one overlap since this will give Dirichlet condition for the copy (?). I think the problem is that of some reason the residual is only calculated for the owner cells since this is the only needed for the original solvers. With this copying and the additional setting of zero for the copy attribute (i.e. "useless overlap") the right hand side is correct. The correct thing is to assemble for all owner and overlap (I would have prefered all) and then set it to zero (at the same time as the matrix elements is fixed to get correct Dirichlet condition). If amg solver is used I think a change in the redistribution function is needed to get correct behavoir if the coarse scale communicator has overlap attributes.